packages/block-library/src/post-featured-image/dimension-controls.js

```diff
@@ -33,14 +33,23 @@ const SCALE_OPTIONS = (
 		/>
 		<SegmentedControlOption
 			value="fill"
-			label={ _x(
-				'Stretch',
-				'Scale option for Image dimension control'
-			) }
+			label={ _x( 'Fill', 'Scale option for Image dimension control' ) }
 		/>
 	</>
 );
 
+const scaleHelp = {
+	cover: __(
+		'Image is sized to maintain its aspect ratio and cropped to fill the entire space.'
```

Comment:
The mention of aspect ratio could be confusing as I'd consider that width/height of the natural image.

How about something like:
'Image is cropped and sized to fill the entire space without being distorted'